ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ClassicProtocolSpecs.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.mainnet;
+
+import org.hyperledger.besu.ethereum.core.Account;
+import org.hyperledger.besu.ethereum.core.Address;
+import org.hyperledger.besu.ethereum.mainnet.contractvalidation.MaxCodeSizeRule;
+import org.hyperledger.besu.ethereum.vm.MessageFrame;
+
+import java.math.BigInteger;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import com.google.common.collect.ImmutableSet;
+
+public class ClassicProtocolSpecs {
+
+  public static final int SPURIOUS_DRAGON_CONTRACT_SIZE_LIMIT = 24576;
+
+  private static final Address RIPEMD160_PRECOMPILE =
+      Address.fromHexString("0x0000000000000000000000000000000000000003");
+
+  // A consensus bug at Ethereum mainnet transaction 0xcf416c53
+  // deleted an empty account even when the message execution scope
+  // failed, but the transaction itself succeeded.
+  private static final ImmutableSet<Address> SPURIOUS_DRAGON_FORCE_DELETE_WHEN_EMPTY_ADDRESSES =
+      ImmutableSet.of(RIPEMD160_PRECOMPILE);
+
+  public static ProtocolSpecBuilder<Void> tangerineWhistleDefinition(
+      final Optional<BigInteger> chainId,
+      final OptionalInt contractSizeLimit,
+      final OptionalInt configStackSizeLimit) {
+    return MainnetProtocolSpecs.homesteadDefinition(contractSizeLimit, configStackSizeLimit)
+        .gasCalculator(TangerineWhistleGasCalculator::new)
```

Comment:
Tangerine Whistle and Spurious Dragon occurred at the same time?  If there is no ETC name for this combined fork then it should be called out in a code comment.